datameta/api/users.py

```diff
@@ -68,6 +68,37 @@ def get_whoami(request: Request) -> UserResponseElement:
     )
 
 
+@view_config(
+    route_name="user_id",
+    renderer="json",
+    request_method="GET",
+    openapi=True
+)
+def get(request: Request):
+
+    # Authenticate the user
+    auth_user = security.revalidate_user(request)
+
+    user_id = request.matchdict["id"]
+    db = request.dbsession
+
+    # Get the targeted user
+    target_user = resource_by_id(db, User, user_id)
+
+    if not authz.view_user(auth_user, target_user):
+        raise HTTPNotFound()
+
+    return UserResponseElement(
+        id              =   get_identifier(target_user),
+        name            =   target_user.fullname,
+        group_admin     =   target_user.group_admin,
+        site_admin      =   target_user.site_admin,
+        site_read       =   target_user.site_read,
+        email           =   target_user.email,
```

Comment:
Let's include the permission related and `email` fields only for site admins

Reply:
I am currently using the `UserResponse` for this, which doesn't allow the fields to be optional. Do you want me to create a new Response class or should I just set email to empty string and the perm-attribs to False?

Reply:
Rename the old one to `WhoamiResponse` and create a new `UserResponse`.